mautic/mautic
# 9704-fix-List-Unsubscribe

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.3.0
| Bug fix?                               | yes
| New feature?                           |no
| Automated tests included?              | no
| Issue(s) addressed                     | Fixes #9704 



#### Description:
this fix should resolve #9704 . before fix emails has wrong unsubscribe link in email header - every batch gets one link of the last contact in the batch. reason of this behavior is the fact that setMessageHeaders() called once a batch. solution - use token instead of static value


#### Steps to test this PR:

    1. create a campaign with more than 1 receipt.
    2. open the email received and check original message.
    3. on the header u should find List-Unsubscribe that would contain the right unsubscribe link (make sure u checking more than one email)
